Doc/library/shutil.rst

```diff
@@ -65,6 +66,13 @@ Directory and files operations
    a new symbolic link will be created instead of copying the
    file *src* points to.
 
+   .. note::
+      Internally platform-specific zero-copy syscalls are used on Linux, OSX
```

Comment:
With the length of this note, and the number of functions it applies to, I'm thinking it may make sense to add a new subsection before the `copytree` and `rmtree` examples called "Platform-dependent efficient copy operations", that provides some more details on the underlying operations used, and the fallback to `copyfileobj`.

Then the various `versionchanged` notes can just link to that new subsection.

This will also give room to explain that the optimisation doesn't apply to all platforms that offer `os.sendfile`, it's specifically limited to those that allow the destination of `sendfile` to be a regular file descriptor (not those that require it to be an open socket stream).

Reply:
Good idea. Done.